pkg/webapi/webapi.go

```diff
@@ -38,6 +36,12 @@ func StateHandler(pool *trcapi.Pool) http.HandlerFunc {
 			return
 		}
 
+		//Check if the authentication token matches
+		if trcConn.Token != r.Header.Get("token") {
+			http.Error(w, fmt.Sprintf("Wrong authentication token"), http.StatusUnauthorized)
+			trcConn.Close()
```

Comment:
Connection should not be closed here, also token should be checked ASAP before doing anything else, we want to fail early.(applies to other endpoints as well)